monetio/sat/_tempo_l2_no2_mm.py

```diff
@@ -0,0 +1,146 @@
+"""TEMPO L2 NO2 data file reader.
+
+History:
+
+
+"""
+
+import logging
+import os
+import sys
+from collections import OrderedDict
+from glob import glob
+from pathlib import Path
+
+import numpy as np
+import xarray as xr
+from netCDF4 import Dataset
+
+
+def _open_one_dataset(fname, variable_dict):
+    """Read locally stored INTELSAT TEMPO NO2 level 2
+    Parameters
+    ----------
+    fname: string
+        fname is local path to netcdf4 (HDF5) file
+
+    variable_dict: dict
+    Returns
+    -------
+    ds: xarray dataset
+    """
+    print("reading " + fname)
+
+    ds = xr.Dataset()
+
+    dso = Dataset(fname, "r")
+    lon_var = dso.groups["geolocation"]["longitude"]
+    lat_var = dso.groups["geolocation"]["latitude"]
+    time_var = dso.groups["geolocation"]["time"]
+
+    ds["lon"] = (
+        ("x", "y"),
+        lon_var[:].squeeze(),
+        {"long_name": lon_var.long_name, "units": lon_var.units},
+    )
+    ds["lat"] = (
+        ("x", "y"),
+        lat_var[:].squeeze(),
+        {"long_name": lat_var.long_name, "units": lat_var.units},
+    )
+    ds["time"] = (
+        ("time",),
+        time_var[:].squeeze(),
+        {"long_name": time_var.long_name, "units": time_var.units},
+    )
+    ds["lon"] = ds["lon"].fillna(-999.99)
+    ds["lat"] = ds["lat"].fillna(-999.99)
```

Comment:
Why not leave them as NaN?

Reply:
Because matplotlib.pcolormesh breaks if there are NaN values in the coordinates (the _values_ can be NaN, but not the coordinates). However, he lat/lon that are NaN in the original arrays are because the satellite is pointing at space, and therefore lat and lon are not defined.

 Another solution might be using contourf, but that would break away from the behavior of draw_map() in MONET. 
The workaround of turning them into -999.99 was taken from Amy K. Huff (https://gist.github.com/akhuff157/8411788ed06d459972f9b15203c8d793), but if you have a better solution, I am happy to change this.

Reply:
Since this issue is more to do with plotting, I would prefer this not be done in the reader. Perhaps it could be done in MM before pcolormesh is used. Or, we could do it here but make it an option that is off by default.

Reply:
Let's do it at the MM level.
I removed those lines in [fb4f571](https://github.com/noaa-oar-arl/monetio/pull/188/commits/fb4f571e86dbd3348b61eeb6904435a335c6d5d0)